src/cli/lint/internalLints/upgrade/patches/10.1.0/migrateNpmrcToPnpmWorkspace.ts

```diff
@@ -0,0 +1,182 @@
+import { inspect } from 'util';
+
+import { glob } from 'fast-glob';
+import { promises as fs } from 'fs-extra';
+
+import type { PatchFunction, PatchReturnType } from '../..';
+import {
+  findCurrentWorkspaceProjectRoot,
+  findWorkspaceRoot,
+} from '../../../../../../utils/dir';
+import { log } from '../../../../../../utils/logging';
+import { hasNpmrcSecret } from '../../../../../../utils/npmrc';
+import { replaceManagedSection } from '../../../../../configure/processing/configFile';
+
+const NPMRC = '.npmrc';
+
+const checkFileExists = async (filePath: string) => {
+  try {
+    await fs.access(filePath);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+const migrateCustomNpmrcSettings = async () => {
+  const contents = await fs.readFile(NPMRC, 'utf-8');
+
+  const remainderLines = replaceManagedSection(contents, '')
+    .split('\n')
+    .map((line) => line.trim())
+    .filter((line) => line.length > 0)
+    .filter((line) => !line.startsWith('#'))
+    .filter((line) => !hasNpmrcSecret(line));
+
+  if (remainderLines.length === 0) {
+    return;
+  }
+
+  const pnpmWorkspaceFile = 'pnpm-workspace.yaml';
+  const pnpmWorkspaceExists = await checkFileExists(pnpmWorkspaceFile);
+  if (!pnpmWorkspaceExists) {
+    await fs.writeFile(pnpmWorkspaceFile, '');
+  }
+
+  // prepend the lines to the pnpm-workspace.yaml file, but commented out
+  const pnpmWorkspaceContents = await fs.readFile(pnpmWorkspaceFile, 'utf-8');
+  const commentedLines = remainderLines.map((line) => `# ${line}`).join('\n');
+  const newContents = `# TODO: Translate these settings to the required format for pnpm-workspace.yaml.
+# skuba moved these from .npmrc, but doesn't know what they mean.
+# See: https://pnpm.io/settings
+#
+${commentedLines}
+
+${pnpmWorkspaceContents}`;
+
+  await fs.writeFile(pnpmWorkspaceFile, newContents);
+};
+
+const fixDockerfiles = async () => {
+  const fileNames = await glob(['**/Dockerfile*']);
+
+  await Promise.all(
+    fileNames.map(async (fileName) => {
+      const contents = await fs.readFile(fileName, 'utf8');
+      const patched = contents.replaceAll(
+        /--mount=type=bind,source=.npmrc,target=.npmrc/g,
```

Comment:
Do we need the regex here?

```suggestion
        '--mount=type=bind,source=.npmrc,target=.npmrc',
```